static/js/ducks/instruments.js

```diff
@@ -13,6 +17,13 @@ export const fetchInstruments = () =>
 export const fetchInstrumentForms = () =>
   API.GET(`/api/internal/instrument_forms`, FETCH_INSTRUMENT_FORMS);
 
+// Websocket message handler
+messageHandler.add((actionType, payload, dispatch) => {
+  if (actionType === REFRESH_INSTRUMENTS) {
```

Comment:
Great. Now you have to make sure this message is actually being sent, by adding a `self.push_all(action='skyportal/REFRESH_INSTRUMENTS)` to the instrument.py handler, for POST, PUT and DELETE. This means if you add/change/remove an instrument using the API, the front end will know about this. 

On the first version of the test you had for this page you posted an instrument via API. I suggest you return that test (besides the front end input you have) and make sure the new instrument appears without a page refresh. That way you'll know you closed the loop.

Reply:
Ah yes, added now. And that instrument API call is still there, nominally that is what that filter check is doing.